assemble/Momentum_DG.F90

```diff
@@ -314,10 +321,6 @@ subroutine construct_momentum_dg(u, p, rho, x, &
        end if
        have_advection = .true.
     else
-       ! Forcing a zero NonlinearVelocity will disable advection.
-       call allocate(U_nl, U%dim,  U%mesh, "NonlinearVelocity", &
-            FIELD_TYPE_CONSTANT)
-       call zero(U_nl)
        have_advection=.false.
        advecting_velocity => U_nl
```

Comment:
With advection_scheme/none, we now have advecting_velocity pointing to NonlinearVelocity instead of a zeroed vector field. Are we sure this is not going to change things (i.e. is advecting_velocity not used when have_advection==.false.)?